HW4.py

```diff
@@ -0,0 +1,316 @@
+    y = data.target
+
+    # Determine if response is continuous or boolean
+
+    if y.dtype is str or bool is True:
+        response_type = "boolean"
+        print("---Response is boolean---")
+    elif np.unique(y).size / y.size < 0.05:
+        response_type = "boolean"
+        print("---Response is boolean---")
+    else:
+        response_type = "continuous"
+        print("---Response is continuous---")
+
+    # Determine if the predictor continuous or boolean &
+    # create plots for each variable type
+
+    predictor_type = []
+    for idx, column in enumerate(X.T):
+        feature_name = data.feature_names[idx]
+
+        predictor = statsmodels.api.add_constant(column)
+        # Get the stats & plot
+        if column.dtype is str or bool is True:
+            v_type = "boolean"
+            print(data.feature_names[idx], "is boolean")
+            if response_type == "continuous":
+
+                logistic_regression_model = statsmodels.api.GLM(y, predictor)
+                logistic_regression_model_fitted = (
+                    logistic_regression_model.fit()
+                )  # noqa
+                print(f"Variable: {feature_name}")
+                print(logistic_regression_model_fitted.summary())
+                t_value = round(logistic_regression_model_fitted.tvalues[1], 6)
+                p_value = "{:.6e}".format(
+                    logistic_regression_model_fitted.pvalues[1]
+                )  # noqa
+
+                # Categorical Predictor by Continuous Response
+
+                fig = px.scatter(x=column, y=y)
+                fig.update_layout(
+                    title=f"Variable: {feature_name}: (t-value={t_value}) (p-value={p_value})",  # noqa
+                    xaxis_title=f"Variable: {feature_name}",
+                    yaxis_title="y",
+                )
+                fig.show()
+
+            else:
+                print(data.feature_names[idx], "is boolean")
+
+                logistic_regression_model = statsmodels.api.GLM(y, predictor)
+                logistic_regression_model_fitted = (
+                    logistic_regression_model.fit()
+                )  # noqa
+                print(f"Variable: {feature_name}")
+                print(logistic_regression_model_fitted.summary())
+                t_value = round(logistic_regression_model_fitted.tvalues[1], 6)
+                p_value = "{:.6e}".format(
+                    logistic_regression_model_fitted.pvalues[1]
+                )  # noqa
+
+                # Continuous Predictor by Continuous Response
+
+                fig = px.scatter(x=column, y=y)
+                fig.update_layout(
+                    title=f"Variable: {feature_name}: (t-value={t_value}) (p-value={p_value})",  # noqa
+                    xaxis_title=f"Variable: {feature_name}",
+                    yaxis_title="y",
+                )
+                fig.show()
+
+        elif round((np.unique(X.T[idx]).size / X.T[idx].size), 2) <= 0.05:

```

Comment:
I think this might run into an issue with datasets that have continuous values that happen to not be very unique. Like a dataset of percentages where all but a few values are 100 or 0 could end up with maybe 5 unique values in 100s of rows. But it should still treat them as continuous.

Reply:
I am thinking which is the good method to filter this situation out

Reply:
I like how you think Michelle. I'm not a fan of this method either,... but I'm not going to count it against anyone.